src/main/java/software/amazon/event/kafkaconnector/logging/GitCommitAwareLoggerFactory.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package software.amazon.event.kafkaconnector.logging;
+
+import java.io.IOException;
+import java.util.Objects;
+import java.util.Properties;
+import org.apache.kafka.common.utils.LogContext;
+import org.slf4j.Logger;
+
+public class GitCommitAwareLoggerFactory {
```

Comment:
nit: would `GitCommitLoggerFactory` also be ok?

Reply:
That's fine, but misleading in my view because it sounds like only the Git commit is logged.

Reply:
Fair point.

Reply:
I'm fine with `GitAwareLoggerFactory`.

Reply:
Q: Do we want to bind this to a concrete implementation? Whenever we want to add new information to the log we would need to change all definitions of loggers across the code base. This might be unlikely but still since we already have this concept of LogContext - wouldn't `ContextAwareLoggerFactory` be sufficient? (Whatever the Log context then might be).

Reply:
`ContextAwareLoggerFactory` is better, will change to it.